modules/creation/adding-configuration-page-modern.md

```diff
@@ -0,0 +1,489 @@
+}
+```
+
+{{% notice note %}}
+More informations about controllers on [Controller & routing]({{<relref "/8/development/architecture/modern/controller-routing">}}) and [Symfony official documentation](https://symfony.com/doc/4.4/controller.html)
+{{% /notice %}}
+
+### Create a route for the configuration controller
+
+Create a `routes.yml` file in `config/`. 
+
+```yml
+demo_configuration_form:
+  path: /demosymfonyformsimple/configuration
+  methods: [GET, POST]
+  defaults:
+    _controller: 'PrestaShop\Module\DemoSymfonyFormSimple\Controller\DemoConfigurationController::index'
+    # Needed to work with tab system
+    _legacy_controller: AdminDemoSymfonyFormSimple
+    _legacy_link: AdminDemoSymfonyFormSimple
+```
+
+{{% notice note %}}
+More informations about routing on [Controller & routing]({{<relref "/8/development/architecture/modern/controller-routing">}}) and [Symfony official documentation](https://symfony.com/doc/4.4/routing.html)
+{{% /notice %}}
+
+## Add this route to the getContent() method of the module
+
+The `getContent()` method of a module is the method called when accessing the configuration page of the module. 
+
+Add this method to your module, with a redirection to the route previously registered.
+
+```php
+  public function getContent()
+  {
+      $route = SymfonyContainer::getInstance()->get('router')->generate('demo_configuration_form');
+      Tools::redirectAdmin($route);
+  }
+```
+
+## Install and enable your module
+
+Go to the Back Office, and in Modules, find your module, then install and enable it. 
+
+{{% notice note %}}
+You can also install it via CLI: 
+
+```
+php bin/console prestashop:module install demosymfonyformsimple
+php bin/console prestashop:module enable demosymfonyformsimple
+```
+{{% /notice %}}
+
+## Navigate / test your configuration form
+
+Open your browser, and navigate to `Back Office > Modules`.
+
+Then, find your module (`Demo symfony form configuration`), and click on the `Configure` button. 
+
+## Complete example module
+
+The module created in this guide is available [here](https://github.com/PrestaShop/example-modules/tree/master/demosymfonyformsimple).
+
+## Other form types
+
+You can use all [native Symfony Form Types](https://symfony.com/doc/4.4/reference/forms/types.html), and the PrestaShop specific ones: [see Form Types reference]({{<relref "/8/development/components/form/types-reference">}}).
+
+{{% notice note %}}
+Please refer to this example module for a complete implementation of those fields: [DemoSymfonyForm](https://github.com/PrestaShop/example-modules/tree/master/demosymfonyform)
+{{% /notice %}}
+
+### Required Javascript for some types
+
+Some types requires Javascript initializations to properly work. 
+
+Add a JS file to your form's twig template: 
+
+```html
+{% block javascripts %}
+  {{ parent() }}
+  <script src="{{ asset('../modules/demosymfonyform/views/js/form.js') }}"></script>
+{% endblock %}
+```
+
+And add the Javascript code to this JS file:
```

Comment:
```suggestion
And add the following JavaScript code to this file:
```

Reply:
It's great that you mention it, but there's no direct correlation between your guide and this. Maybe if you limit the field length to 32 chars and use `TextWithLengthCounter`?